components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+package org.wso2.carbon.apimgt.rest.api.admin.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowException;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowExecutor;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowExecutorFactory;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowStatus;
+import org.wso2.carbon.apimgt.rest.api.admin.WorkflowsApiService;
+import org.wso2.carbon.apimgt.rest.api.admin.dto.WorkflowDTO;
+import org.wso2.carbon.apimgt.rest.api.util.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+
+/**
+ * This class is contains rest apis related to workflows
+ */
+public class WorkflowsApiServiceImpl extends WorkflowsApiService {
+
+    private static final Log log = LogFactory.getLog(WorkflowsApiService.class);
+
+    /**
+     * This is used to update the workflow status
+     *
+     * @param workflowReferenceId workflow reference id that is unique to each workflow
+     * @param body                body should contain the status, optionally can contain a
+     *                            description and an attributes object
+     * @return
+     */
+    @Override
+    public Response workflowsUpdateWorkflowStatusPost(String workflowReferenceId, WorkflowDTO body) {
+        ApiMgtDAO apiMgtDAO = ApiMgtDAO.getInstance();
+        boolean isTenantFlowStarted = false;
+
+        try {
+            if (workflowReferenceId == null) {
+                RestApiUtil.handleBadRequest("workflowReferenceId is empty", log);
+            }
+
+            org.wso2.carbon.apimgt.impl.dto.WorkflowDTO workflowDTO = apiMgtDAO.retrieveWorkflow(workflowReferenceId);
```

Comment:
Shall we move the import statements to the top rather than using inline imports?